File: medium/us_date/us_date.kt
fun date_format(date:String):String{
    
    var date_output:List<String> = mutableListOf()
    
    if("/" in date){
        
        date_output = date.split("/")
        return(date_output[1] + "/" + date_output[0] + "/" + date_output[2])
        
    }else{
        var months = mapOf(Pair("January", "1"),
                           Pair("February", "2"),
                           Pair("March", "3"),
                           Pair("April", "4"),
                           Pair("May", "5"),
                           Pair("June", "6"),
                           Pair("July", "7"),
                           Pair("August", "8"),
                           Pair("September", "9"),
                           Pair("October", "10"),
                           Pair("November", "11"),
                           Pair("December", "12"))
        
        date_output = date.split(" ")
        return(date_output[1].substring(0,date_output[1].length-1) + "/" + months[date_output[0]] + "/" + date_output[2])
    }
}	


fun main(args: Array<String>){
	
    println("Date : ")
    //var date = readLine()
  
    println(date_format("November 19, 2019"))     
   
  
}

File: medium/us_date/us_date.rb
"""
@author	André Peralta
@version 20/05/2021
"""

def USdate date

	months = {'January'=> '1',
		'February'=> '2',
		'March'=> '3',
		'April'=> '4',
		'May'=> '5',
		'June'=> '6',
		'July'=> '7',
		'August'=> '8',
		'September'=> '9',
		'October'=> '10',
		'November'=> '11',
		'December'=> '12'}

	if date.index("/")
		dateList = date.split("/")
		return dateList[1] + "/" + dateList[0] + "/" + dateList[2]
	else
		dateList = date.split(" ")
		return dateList[1][0] + "/" + months[dateList[0]] + "/" + dateList[2]

	end
end


puts "Date: "
date = gets

print USdate(date)

